pennylane/compiler/qjit_api.py

```diff
@@ -105,8 +106,177 @@ def circuit(theta):
     """
 
     if not available(compiler):
-        raise RuntimeError(f"The {compiler} package is not installed.")
+        raise CompilerNotFoundError(f"The {compiler} package is not installed.")
 
     compilers = AvailableCompilers.names_entrypoints
     qjit_loader = compilers[compiler]["qjit"].load()
     return qjit_loader(fn=fn, *args, **kwargs)
+
+
+def while_loop(*args, compiler="catalyst", **kwargs):
+    """A :func:`~.qjit` compatible while-loop decorator for PennyLane programs.
+
+    .. note::
+        This function is a wrapper around :func:`catalyst:.while_loop`.
+
+        Please see the Catalyst :doc:`quickstart guide <catalyst:dev/quick_start>`,
+        as well as the :doc:`sharp bits and debugging tips <catalyst:dev/sharp_bits>`
+        page for an overview of the differences between Catalyst and PennyLane.
```

Comment:
```suggestion

        When used with :func:`~.qjit`, this function only supports
        the Catalyst compiler. Please see
        :func:`~catalyst:catalyst.while_loop` for more details.

        Please see the Catalyst :doc:`quickstart guide <catalyst:dev/quick_start>`,
        as well as the :doc:`sharp bits and debugging tips <catalyst:dev/sharp_bits>`
        page for an overview of the differences between Catalyst and PennyLane.
```

Reply:
I think better to phrase it this way, then to call it a wrapper, because what if someone provides another compiler with While support!

Reply:
Thanks!